agent-core/src/main/java/com/newrelic/agent/android/harvest/HarvestConnection.java

```diff
@@ -201,6 +202,24 @@ public HarvestResponse sendData(Harvestable harvestable) {
         return send(dataPost, harvestable.toJsonString());
     }
 
+    /**
+     * Perform a {@code data} service call to the collector and return its {@link HarvestResponse}.
+     *
+     * @param harvestable as string
+     * @return The {@link HarvestResponse} from the collector {@code data} call.
+     */
+    public HarvestResponse sendData(String harvestable) {
```

Comment:
Why was this added to HarvestConnection()?

Reply:
The first sendData is passed in with Harvestable object, which then later on being converted to string. When reading from local file, it's easier to pass in as string directly to be sent over server.